### ISSUE TITLE
problems matching Windows executables

### ISSUE THREAD

**Issue by overlinden** [j256/simplemagic#5]
Hi,

I am writing a file analyser in java and plan to use your simplemagic application to determine the file type of an input stream. To use your application as comfortable as possible I compiled it into a jar file including your magic.gz file.

Unfortunately the findMatch Method always returns null for all types on inputstreams. I tried windows executables, bitmaps, plain text, ...
Is there any error in the current version of your code or do I just call it in a wrong way?

This is my test code:
```java
        ContentInfoUtil util = new ContentInfoUtil();
        ContentInfo info = util.findMatch(new FileInputStream(new File("C:\\test\\bmp.bmp")));
        System.out.println(info != null ? info.getMessage() : "Is null");
```

You can see the folder structure of the simplemagic.jar file attached.
The com folder includes the class files of your code. The res folder includes the magic.gz dictionary file.
![simplemagic](https://cloud.githubusercontent.com/assets/5543041/3090319/c46cea98-e58a-11e3-8ccf-ba86a94aff27.JPG)

Thank you for your help.

**Comment by swpalmer** [comment]
Note that the unit test for .exe files is broken.
ContentInfoUtilTest.java has:

					new FileType("/files/x.exe", ContentType.OTHER, "MythTV", null,
							"MythTV NuppelVideo v (640x480),progressive,aspect:1.00,fps:29.97"),

which is clearly a copy and paste error from a few lines above.

I noticed this after fixing another error that was causing the test to fail on "/files/x.gz".  After fixing that issue, the bogus test for x.exe started failing (as it should have been all along).

The problem that caused the x.gz test to fail was in the method findWhitespaceWithoutEscape.   It needed the following else clause added to the if statement:
 else {
                lastEscape = false; // don't leave escape active after non-whitespace escaped character
 }

**Comment by j256** [comment]
This has been fixed in version 1.7.  Turned out that I was not handling some specific magic file line patterns correctly.
